report/src/main/webapp/app/services/rerun.service.js

```diff
@@ -0,0 +1,130 @@
+/*
+  * AET
+  *
+  * Copyright (C) 2013 Cognifide Limited
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *      http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+ define(['angularAMD', 'endpointConfiguration', 'suiteInfoService' , 'caseFactory'], function (angularAMD) {
+  'use strict';
+  angularAMD.factory('rerunService', rerunService);
+
+  /**
+   * Service responsible for fetching suite's history
+   */
+  function rerunService($rootScope, $http, endpointConfiguration, suiteInfoService, caseFactory) {
+    $rootScope.endpointUrl = endpointConfiguration.getEndpoint().getUrl;
+    var service = {
+      getRerunStatistics: getRerunStatistics,
+      rerunSuite: rerunSuite,
+      rerunTest: rerunTest,
+      rerunURL: rerunURL,
+    };
+
+    return service;
+
+    function getRerunStatistics(msg) {
+      var statsArray = msg;
+      statsArray = statsArray.replace(/:::/, ',');
+      statsArray = statsArray.replace(/\d{2}:\d{2}:\d{2}\.\d{3}/g, '');
+      statsArray = statsArray.replace(/[a-zA-z]|:/g, '');
+      statsArray = statsArray.replace(/\s/g, '');
+      statsArray = statsArray.split(',');
+      return statsArray;
+    }
+
+    function rerunSuite() {
+      $rootScope.rerunInProgress = true;
+      var suiteInfo = suiteInfoService.getInfo();
+      var rerunParams = 'company=' + suiteInfo.company + '&' + 'project=' + suiteInfo.project + '&' +
+        'suite=' + suiteInfo.name;
+      var url = endpointConfiguration.getEndpoint().getUrl + 'suite-rerun?' + rerunParams;
+      $http.post(url, {}).then(function successCallback(response) {
+        $rootScope.rerunMsg = 'Suite rerun initialized';
+        $rootScope.rerunInProgressSuccessful = true;
+        $rootScope.rerunProgress = 0;
+        checkRerunStatus(response.data.statusUrl);
+      }, function errorCallback(response) {
+        $rootScope.rerunMsg = response.statusText;
+      });
+    }
+
+    function rerunTest(testName) {
+      $rootScope.rerunInProgress = true;
+      var suiteInfo = suiteInfoService.getInfo();
+      var rerunParams = 'company=' + suiteInfo.company + '&' + 'project=' + suiteInfo.project + '&' +
+        'suite=' + suiteInfo.name + '&' + 'testName=' + testName;
+      var url = endpointConfiguration.getEndpoint().getUrl + 'suite-rerun?' + rerunParams;
+      console.log(url);
```

Comment:
Please remove `console.log`, this shouldn't be used in production ready code in that form.